bindings/pyroot/ROOT.py

```diff
@@ -238,6 +240,86 @@ def _TTree__iter__( self ):
 
 _root.CreateScopeProxy( "TTree" ).__iter__    = _TTree__iter__
 
+# Array interface
+def _proxy__array_interface__(self):
+    getter_array_interface = "_get__array_interface__"
+    if hasattr(self, getter_array_interface):
+        return self._get__array_interface__()
+    else:
+        raise Exception("Class {} does not have method {}.".format(
+            type(self), getter_array_interface))
+
+for pyclass in [
+        "std::vector<{dtype}>",
+        "Experimental::VecOps::TVec<{dtype}>"
+        ]:
+    dtypes = ["float", "double", "int", "unsigned int", "long", "unsigned long"]
+    for dtype in dtypes:
+        class_scope = _root.CreateScopeProxy(pyclass.format(dtype=dtype))
+        class_scope._proxy__array_interface__ = _proxy__array_interface__
+        class_scope.__array_interface__ = property(class_scope._proxy__array_interface__)
+
+# TTree.AsMatrix functionality
+def _TTreeAsMatrix(self, columns=None, dtype="double"):
+    # Import numpy lazily
+    try:
+        import numpy as np
+    except:
+        raise ImportError("Failed to import numpy during call of TTree.AsMatrix.")
+
+    # Check that tree has entries
+    if self.GetEntries() == 0:
+        raise Exception("Tree {} has no entries.".format(self.GetName()))
+
+    # Get all columns of the tree if no columns are specified
+    if columns is None:
+        columns = [branch.GetName() for branch in self.GetListOfBranches()]
+
+    # Check that all given columns exist
+    for col in columns:
+        branch = self.GetBranch(col)
+        if branch == None:
+            raise Exception("Tree {} has no branch {}.".format(self.GetName(), col))
+
+    # Check that given data-type is supported
+    supported_dtypes = ["int", "unsigned int", "long", "unsigned long", "float", "double"]
+    if not dtype in supported_dtypes:
+        raise Exception("Data-type {} is not supported, select from {}.".format(
+            dtype, supported_dtypes))
+
+    # Convert columns interable to std.vector("string")
+    columns_vector = _root.std.vector("string")(len(columns))
+    for i, col in enumerate(columns):
+        columns_vector[i] = col
+
+    # Get data-types of columns
+    col_dtypes = [self.GetLeaf(col).GetTypeName() for col in columns]
```

Comment:
@dpiparo @etejedor In the case for `columns=None`, I select all branches as columns. This is done here without error-handling. Either we leave it like this or I add a hard-coded white-list with arithmetic data-types in ROOT. Which option do you prefer?